### DISCUSSION TITLE
Migration from OpenZWave on Domoticz into an docker image with zwavejs-ui

### DISCUSSION THREAD

**Discussion by darksartor** [zwave-js/zwave-js-ui#3797] [Q&A] [answered]
Dear support,

I try to setup correctly my Z‐Stick Gen5 USB Controller on a ZWaveJS-UI docker image but many points dont work fine:
- How setup correctly all Security Keys ? Before on OpenZWave I have only 1 security key setuped and now the ask me... 6 ? 

![SecurityKeys](https://github.com/zwave-js/zwave-js-ui/assets/15523464/2ce0cd80-7d81-4592-8dd2-eb396063bb6b)

Must I put same key everywhere ? 

-Look red message like "Node 021 supports Security S0, but the S0 network key was not configurated ..." but when I try to fix it, I obtain other error about configuration... ???

![SecurityKeys2](https://github.com/zwave-js/zwave-js-ui/assets/15523464/9c8a5662-894a-4d08-a92d-7e911aed0069)

Other question, I try to plug a MQTT Explorer outside on the ZWaveJS-UI docker to verify than all device are available but It don't open ECONNRESET... why ? 

NB: I reroot port 1884 -> 1883

![MQTTExplorer](https://github.com/zwave-js/zwave-js-ui/assets/15523464/2ff4e120-452f-4a93-94cb-ddc51fd6ae3c)

![DockerPort](https://github.com/zwave-js/zwave-js-ui/assets/15523464/565f5d2d-87e5-4a19-a21a-b8bceeaded5a)

And finally I can't link correctly from my Domoticz docker to link at all ZWave device from ZWaveJS-UI docker image (I'm sure if I solve MQTT Explorer it will also solve this problem)

![DomoticzLinkMQTT](https://github.com/zwave-js/zwave-js-ui/assets/15523464/abccdbef-4e9f-46dd-b60d-154642cd5926)

I don't see any ZWave device ... 

Thanks for your help !

**Comment by robertsLando** [comment, marked as answer]
> How setup correctly all Security Keys ? Before on OpenZWave I have only 1 security key setuped and now the ask me... 6 ?

Because OZW didn't have support for S2 and S2 requires others security keys. You can just copy paste your old key in S0 key and generate random ones in the others (by using the button on the right side of them)

> Other question, I try to plug a MQTT Explorer outside on the ZWaveJS-UI docker to verify than all device are available but It don't open ECONNRESET... why ?

ZUI is an mqtt client and it publish data to a broker, did you spawn a broker on port 1883? If not I suggest you to add mosquitto to your docker compose file (or run it locally)

> And finally I can't link correctly from my Domoticz docker to link at all ZWave device from ZWaveJS-UI docker image (I'm sure if I solve MQTT Explorer it will also solve this problem)

AFAIK domoticz uses mqtt discovery so firstly you should fix the mqtt issue described above by adding your broker and then remember to enable mqtt discovery on settings

**Comment by darksartor** [comment]
Dear all,

FINALLY is working quiet fine I could correctly setup Mosquitto MQTT Broker and receive all device. The problem was fact than we need setup port for listener 1883 (and let anonymous access to) in MQTT config file mosquitto.conf AND edit the good config file (many was on my Synology and good config file wasn't taked) Without this, only local (into container) could connect to the internal listener... :roll:

But I have another problem... from container Domoticz I could see all my zwave devices and status, but I couldn't update it (switch ON/OFF) only from ZWave-JS-UI. I don't know how and after some manipulation I could let to control zwave switch device from Domoticz... but don't work anymore I surely disable something... but what ? Someone know how to control device from Domoticz container ? Which setup must I be careful ?

Best regards

**Reply by robertsLando** [reply]
I'm sorry but I never used domoticz so I cannot help you with that. I suggest to open the issue on domoticz repo or forum

**Reply by darksartor** [reply]
Hello, I did it but can I test through a MQTT Explorer If command work to setup Z-Wave device from outside ZWaveJSUI ?

**Reply by robertsLando** [reply]
> Hello, I did it but can I test through a MQTT Explorer If command work to setup Z-Wave device from outside ZWaveJSUI ?

Yes you can, follow docs: https://zwave-js.github.io/zwave-js-ui/#/guide/mqtt?id=set-values

**Comment by darksartor** [comment]
Another question I See we can put name and location to the device globally , but what could be really helpfull it is to setup name also on each Sensor. Exemple a double switch I need to name "switch1" and other "switch2" other else default name is to long and complicated to manipulate on program in Domoticz (or other) and need to be renamed also into. Doing that all name are correctly setuped into zwaveJSUI and stay unique and same each time we repopulate final Domoticz box.

**Reply by robertsLando** [reply]
Actually that's not possible
